service-python/pdfgenerator/src/lib/queues.py

```diff
@@ -57,6 +58,36 @@ def on_fetch_callback(channel, method, properties, body):
     channel.basic_publish(exchange=EXCHANGE, routing_key=properties.reply_to, properties=pika.BasicProperties(correlation_id=properties.correlation_id), body=json.dumps(response))
 
 
+def on_generate_fetch_callback(channel, method, properties, body):
+    try:
+        redis_client = RedisClient(redis_config)
+        pdf_generator = PDFGenerator(pdf_options)
+
```

Comment:
Can you modularize this? It looks like we literally copied two methods here. I would make commonalities functions?

Reply:
Removed the extra function. Added a check in the generate function to decide if it needs to fetch right after.